### PR TITLE
Do not merge yet: Thesaurus

### PR DESCRIPTION
**Pull request by nevrome**
Better material classification.

### CODE REVIEW

**Comment by nevrome** [comment]
Thanks for this excellent work, David. 

Right now we have two different values for missing/bad/unclear information: **NA**  and **undefined**
I suggest to replace **undefined** with **other** and **NA** with **unknown**.

**other:** We have an information, but it's too strange/unique/specific to add it to the other established categories.
**unkown:** We don't have any infomation.

File: data-raw/material_thesaurus.csv [hunk 1]
```diff
@@ -1,249 +1,504 @@
...
+hair,animal hair
+antler,antler
+plant remains,apple
+aragonite,aragonite
+aragonite,aragonte
 ash,ash
+undefined,Ash and charcoal
+undefined,Ash and shell
 ash,ashy soil
+undefined,Austomylitus rostratus shell
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: Why not just "shell"?

File: data-raw/material_thesaurus.csv [hunk 2]
```diff
@@ -1,249 +1,504 @@
...
-bark,riinde
-brain,brain
-brain,brain tissue
+bark,Bark
+plant remains,barley
+NA,bd
+plant remains,bean
+plant remains,beans
+plant remains,been
+undefined,Beeswax
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: own category: beeswax

File: data-raw/material_thesaurus.csv [hunk 3]
```diff
@@ -1,249 +1,504 @@
...
-bone,hueso
-ceramics,pottery
+NA,Br
+shell,Brachidontes rostratus shell
+tissue,brain
+tissue,brain tissue
+food,bread
+NA,Bulk
+NA,bulk
+undefined,Bulk organic fraction
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: maybe we just ignore the "bulk" part and add the to the respective categories

File: data-raw/material_thesaurus.csv [hunk 4]
```diff
@@ -1,249 +1,504 @@
...
+plant remains,burnt plants
+plant remains,Burnt wood
+NA,c
+shell,Cabesatana sp. shell
+calcium carbonate,Calcium carbonate
+undefined,Calcrete nodule
+plant remains,Canarium s.
+NA,carbon
+undefined,Carbon
+undefined,Carbon from charred bone
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: bone?

File: data-raw/material_thesaurus.csv [hunk 5]
```diff
@@ -1,249 +1,504 @@
...
+undefined,Burnt earth from hearth
+plant remains,burnt plants
+plant remains,Burnt wood
+NA,c
+shell,Cabesatana sp. shell
+calcium carbonate,Calcium carbonate
+undefined,Calcrete nodule
+plant remains,Canarium s.
+NA,carbon
+undefined,Carbon
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: charcoal?

File: data-raw/material_thesaurus.csv [hunk 6]
```diff
@@ -1,249 +1,504 @@
...
+undefined,Carbon from charred bone
+NA,carbonate
+undefined,Carbonate
+undefined,Carbonised Eucalypt sp.
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: maybe "carbonized material"

File: data-raw/material_thesaurus.csv [hunk 7]
```diff
@@ -1,249 +1,504 @@
...
+shell,Crassostera commercialis shell
+shell,"Crassostraea sp., Terebralia sp., Telescopium sp., Anadara sp."
+bone,cremated bones
+undefined,Crustal salts
+wood,Dendro
+teeth,dentin
+shell,Dicatgaus orbita shell
+shell,Dicathais sp. shell
+shell,Dicathais textilosa shell
+undefined,Diffuse Charcoal
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: charcoal

File: data-raw/material_thesaurus.csv [hunk 8]
```diff
@@ -1,249 +1,504 @@
...
+freshwater shell,Freshwater mussel (Unio sp.)
+freshwater shell,Freshwater mussel (Velesunio ambiguus sp.)
+freshwater shell,Freshwater mussel shell
+freshwater shell,Freshwater shell
+freshwater shell,Freshwater shell (Unio protovittatus)
+freshwater shell,Freshwater shell (Unionid sp.)
+freshwater shell,Freshwater shell (Velesunio sp.)
+plant remains,fruit
+plant remains,fruit/seeds
+undefined,Gafrarium tumidum shell
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: shell?

File: data-raw/material_thesaurus.csv [hunk 9]
```diff
@@ -1,249 +1,504 @@
...
+undefined,Gastropod shell
+undefined,Geloina coaxans shell
+plant remains,gerste
 plant remains,Gilbertiodendron d.
+undefined,Glass
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: glass! however that worked...

File: data-raw/material_thesaurus.csv [hunk 10]
```diff
@@ -1,249 +1,504 @@
...
+bone,Human bone (water soluble fraction)
+bone,Human bone apatite
+bone,Human bone carbonate
+bone,Human bone collagen
+bone,Human bone collagen amino acids
+bone,Human bone humic acid
+bone,Human bone insolubles
+NA,humates
+soil,humic
+undefined,humic acid
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: own category "humic acid"

File: data-raw/material_thesaurus.csv [hunk 11]
```diff
@@ -1,249 +1,504 @@
...
+plant remains,plant matter
+plant remains,plant Remains
+plant remains,plant remains
+plant remains,plant Remains
+plant remains,Plant Remains
+plant remains,plant seed
+plant remains,plantenvezels
+plant remains,plants
+shell,Plebidonax deltoides shell
+undefined,Pollen (<5 microns)
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: own category "pollen"

File: data-raw/material_thesaurus.csv [hunk 12]
```diff
@@ -1,249 +1,504 @@
...
+undefined,Pollen (10-25 microns)
+undefined,Pollen (45-90 microns)
+undefined,Pollen (5-10 microns)
+undefined,Polymesoda erosa shell
+wood,Populus sp.
+ceramics,pottery
+shell,Pyrazus ebeninus shell
+shell,Pyrazus sp. Shell
+undefined,Quartz
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: own category "quarz"

File: data-raw/material_thesaurus.csv [hunk 13]
```diff
@@ -1,249 +1,504 @@
...
+shell,Turbo undulata shell
+shell,Turbo undulatus and Haliotis rubra shell
+bone,Turtle carapace
+wood,twig
+wood,twigs
+NA,UNKNOWN
+NE,Unknown
+bone,Ur
+soil,veen
+undefined,Vegetable matter lining grave
```
Comment (nevrome) [inline data-raw/material_thesaurus.csv]: plant remains

**Comment by yesdavid** [comment]
Thanks for your comment and your review, Clemens. 
I will apply your suggestions at once.

File: data-raw/material_thesaurus.csv (hunk 3 above)
Comment (yesdavid) [reply data-raw/material_thesaurus.csv]: please elaborate

File: data-raw/material_thesaurus.csv (hunk 4 above)
Comment (yesdavid) [reply data-raw/material_thesaurus.csv]: is there no difference in dating a regular and a charred bone?

**Comment by nevrome** [comment]
Again thanks for this work, David.

I would like to ask the more experienced colleagues (@OliverNakoinz, @MartinHinz, @dirkseidensticker, @nmueller18, @chrinne) to look through it now. Again: The idea of this material classification is to make it more easy to search and sort 14C dates. That's especially important for the material selection in [neolithicRC](https://www.forschungsdatenarchiv.escience.uni-tuebingen.de/cSchmid/neolithicRC/). David prepared [this](https://github.com/ISAAKiel/c14bazAAR/blob/thesaurus/data-raw/material_thesaurus.csv) list of database values and respective classes that should be checked. He documented his work and some important issues in [this](https://github.com/ISAAKiel/c14bazAAR/blob/thesaurus/data-raw/material_thesaurus_comments.md) text file and in [this](https://github.com/ISAAKiel/c14bazAAR/pull/30) PR discussion.

Maybe you could invest 15 minutes to look through the classification and comment on things that should be changed.

**Comment by MartinHinz** [comment]
Some comments:

## list
* aragonite: I am not sure, how important aragonite is, I would have put it under "others".
* ash: IMHO you can not date ash, but only charcoal within it!?
* ceramics: I would not subsume 'Baked clay from heart' and 'mud-brick wall' under ceramics, that category implies a different archaeological class. 'Clayball from hearth' is essentially the same as 'Baked clay from heart'.
* glass: see ash
* charcoal: 'charred wood' and 'Carbonised wood' should be essentially the same, also in case of plant remains. so either both under 'other' or both under 'charcoal'.
* others: 'Mylitus planulatus shell' is mussel. attached to that, if we have 'freshwater shell', we also should have 'saltwater shell'. or dropping the water quality in general, making it 'shell'. Methodological problems are attached to both categories (hardwater effect).
* quartz: see ash
* snail: see comment on shell in others

## not yet defined, uncertain
* the first four entries I would suggest for 'others'
* same for Austomylitus rostratus
* bulk should be others, except it is determined unambiguous (sediment -> soil)
* burnt is important, it implies charcoal
* calcium carbonate -> others?
* carbon I assume charcoal
* differentiation between marine and freshwater shell? I do not think so
* can "glass" even be dated? No
* "human" has been classified as "bone": I would differntiate between human and animals
* "pig" has been classified as "bone": might be good reasons to do so
* what about "human blood protein"? it is organic matter, so it should be datable
* "humid acid" should be one of the few possibilities to directly date soil
* leather: actually, that should be tissue?
* "pollen" as own category? Definitely.
* "stone" has been defined as "soil" -> incorrect? see ash above
* turtle bone should be correct

**Comment by yesdavid** [comment]
thank you for your suggestions, @MartinHinz!

**Pull request merged**